lib/fog/azurerm/models/network/subnet.rb

```diff
@@ -33,6 +41,26 @@ def save
           merge_attributes(Fog::Network::AzureRM::Subnet.parse(subnet))
         end
 
+        def attach_network_security_group(id)
+          subnet = service.attach_network_security_group_with_subnet(resource_group, name, virtual_network_name, address_prefix, route_table_id, id)
+          merge_attributes(Fog::Network::AzureRM::Subnet.parse(subnet))
+        end
+
+        def detach_network_security_group
+          subnet = service.detach_network_security_group_from_subnet(resource_group, name, virtual_network_name, address_prefix, route_table_id)
+          merge_attributes(Fog::Network::AzureRM::Subnet.parse(subnet))
+        end
+
+        def attach_route_table(id)
+          subnet = service.attach_route_table_with_subnet(resource_group, name, virtual_network_name, address_prefix, network_security_group_id, id)
+          merge_attributes(Fog::Network::AzureRM::Subnet.parse(subnet))
+        end
+
+        def detach_route_table
```

Comment:
Only one route table can be attached to subnet. When user invoke this method that one route table is detached.